orc8r/cloud/go/services/ctraced/obsidian/handlers/gateway_api.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handlers
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"magma/orc8r/cloud/go/orc8r"
+	"magma/orc8r/cloud/go/services/configurator"
+	"magma/orc8r/cloud/go/services/dispatcher/gateway_registry"
+	"magma/orc8r/lib/go/protos"
+
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+)
+
+type GWCtracedClient interface {
+	StartCallTrace(networkId string, gatewayId string, req *protos.StartTraceRequest) (*protos.StartTraceResponse, error)
+
+	EndCallTrace(networkId string, gatewayId string, req *protos.EndTraceRequest) (*protos.EndTraceResponse, error)
+}
+
+type GWCtracedClientImpl struct{}
+
+func NewGwCtracedClientImpl() GWCtracedClientImpl {
+	return GWCtracedClientImpl{}
+}
+
+// getGWCtracedClient gets a GRPC client to the ctraced service running on the gateway specified by (network ID, gateway ID).
+// If gateway not found by configurator, returns ErrNotFound from magma/orc8r/lib/go/errors.
+func getGWCtracedClient(networkID string, gatewayID string) (protos.CallTraceServiceClient, context.Context, error) {
+	hwID, err := configurator.GetPhysicalIDOfEntity(networkID, orc8r.MagmadGatewayType, gatewayID)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, fmt.Sprintf("gateway not found, network-id: %s, gateway-id: %s", networkID, gatewayID))
+	}
+	conn, ctx, err := gateway_registry.GetGatewayConnection(gateway_registry.GwCtraced, hwID)
+	if err != nil {
+		errMsg := fmt.Sprintf("gateway ctraced client initialization error: %s", err)
+		glog.Errorf(errMsg, err)
+		return nil, nil, errors.New(errMsg)
+	}
+	return protos.NewCallTraceServiceClient(conn), ctx, nil
+}
+
+// StartCallTrace starts a call trace on the specified gateway
+// If gateway not registered, returns ErrNotFound from magma/orc8r/lib/go/errors.
+func (c GWCtracedClientImpl) StartCallTrace(networkId string, gatewayId string, req *protos.StartTraceRequest) (*protos.StartTraceResponse, error) {
+	client, ctx, err := getGWCtracedClient(networkId, gatewayId)
```

Comment:
Should we check whether the call traceID already exists in blob store and return an error if it does ?

Reply:
Probably not here, this API exists to be fairly dumb